include/native/stream_iteration.h

```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
 
 }
 
+template<typename Generator, typename Operation, typename Predicate, typename Consumer>
+void stream_iteration(parallel_execution_native &ex, Generator && gen, Operation && f, Predicate && condition, Consumer && cons){
+  using namespace std;
+  using generated_type = typename result_of<Generator()>::type;
+  using generated_value_type = typename generated_type::value_type;
+
+  using produced_type = typename result_of<Operation(generated_value_type)>::type;
+  
+  mpmc_queue<generated_type> generated_queue{ex.queue_size,ex.lockfree};
+  mpmc_queue<std::experimental::optional<produced_type>> produced_queue{ex.queue_size,ex.lockfree};
```

Comment:
At the begining of function put 

~~~c++
using namespace std::experimental
~~~

Then you simply use `optional<T>`